### PR TITLE
Add Diffusers Example

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Quick example of how to use the model with Diffusers. It should be faster than the original repo as it uses a faster scheduler (Euler) and optimized fp16.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ PyTorch implementation of InstructPix2Pix, an instruction-based image editing mo
 
 ## TL;DR: quickstart 
 
+To quickly test Intruct-Pix2Pix you can either use the original repository or [🧨 Diffusers](https://github.com/huggingface/diffusers).
+
+
+### Original codebase
```
Comment (patrickvonplaten) [inline README.md:19]:
```suggestion
### Original repository 
```

File: README.md [hunk 2]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+
+prompt = "turn him into cyborg"
+
+images = pipe(prompt, image=image, num_inference_steps=10, image_guidance_scale=1).images
+
+images[0]
+
+```
+
+For more information, dvanced use cases check the docs [here](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/pix2pix)
```
Comment (patrickvonplaten) [inline README.md:98]:
```suggestion
For more information, dvanced use cases check the docs [here](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/pix2pix).
```

File: README.md [hunk 3]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+pipe = StableDiffusionInstructPix2PixPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
+
+pipe.to("cuda")
+
+pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)
+
+url = "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg"
+
+def download_image(url):
+
```
Comment (patrickvonplaten) [inline README.md:79]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+pip install git+https://github.com/huggingface/diffusers.git
+
+```
+
+2. Load the model, an example image and run the model directly in Python:
+
+```python
+
+import PIL
+
```
Comment (patrickvonplaten) [inline README.md:61]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+import requests
+
```
Comment (patrickvonplaten) [inline README.md:63]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+import torch
+
```
Comment (patrickvonplaten) [inline README.md:65]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+from diffusers import StableDiffusionInstructPix2PixPipeline, EulerAncestralDiscreteScheduler
+
+model_id = "timbrooks/instruct-pix2pix"
+
+pipe = StableDiffusionInstructPix2PixPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
+
```
Comment (patrickvonplaten) [inline README.md:71]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+pipe.to("cuda")
+
+pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)
+
+url = "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg"
+
+def download_image(url):
+
+    image = PIL.Image.open(requests.get(url, stream=True).raw)
+
```
Comment (patrickvonplaten) [inline README.md:81]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+    image = PIL.ImageOps.exif_transpose(image)
+
```
Comment (patrickvonplaten) [inline README.md:83]:
```suggestion
```

File: README.md [hunk 10]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+    image = image.convert("RGB")
+
```
Comment (patrickvonplaten) [inline README.md:85]:
```suggestion
```

File: README.md [hunk 11]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+    return image
+
+image = download_image(URL)
+
+prompt = "turn him into cyborg"
+
```
Comment (patrickvonplaten) [inline README.md:91]:
```suggestion
```

File: README.md [hunk 12]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+images = pipe(prompt, image=image, num_inference_steps=10, image_guidance_scale=1).images
+
+images[0]
+
```
Comment (patrickvonplaten) [inline README.md:95]:
```suggestion
```

File: README.md [hunk 13]
```diff
@@ -36,6 +41,62 @@ python edit_app.py
...
+import requests
+
+import torch
+
+from diffusers import StableDiffusionInstructPix2PixPipeline, EulerAncestralDiscreteScheduler
+
+model_id = "timbrooks/instruct-pix2pix"
+
+pipe = StableDiffusionInstructPix2PixPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
```
Comment (patrickvonplaten) [inline README.md:70]:
```suggestion
pipe = StableDiffusionInstructPix2PixPipeline.from_pretrained(model_id, torch_dtype=torch.float16, safety_checker=None)
```

File: README.md [hunk 14]
```diff
@@ -13,6 +13,11 @@ PyTorch implementation of InstructPix2Pix, an instruction-based image editing mo
 
 ## TL;DR: quickstart 
 
+To quickly test Intruct-Pix2Pix you can either use the original repository or [🧨 Diffusers](https://github.com/huggingface/diffusers).
+
+
```
Comment (patrickvonplaten) [inline README.md:18]:
```suggestion
```

File: README.md [hunk 15]
```diff
@@ -36,6 +40,52 @@ python edit_app.py
 
 _(For advice on how to get the best results by tuning parameters, see the [Tips](https://github.com/timothybrooks/instruct-pix2pix#tips) section)._
 
+### 🧨 Diffusers
+
+1. Install diffusers and relevant dependencies:
+
+```bash
+
```
Comment (patrickvonplaten) [inline README.md:48]:
```suggestion
```

File: README.md [hunk 16]
```diff
@@ -36,6 +40,52 @@ python edit_app.py
...
+pip install transformers accelerate torch
+
```
Comment (patrickvonplaten) [inline README.md:50]:
```suggestion
```

File: README.md [hunk 17]
```diff
@@ -36,6 +40,52 @@ python edit_app.py
...
+pip install git+https://github.com/huggingface/diffusers.git
+
```
Comment (patrickvonplaten) [inline README.md:52]:
```suggestion
```

File: README.md [hunk 18]
```diff
@@ -36,6 +40,49 @@ python edit_app.py
...
+url = "https://raw.githubusercontent.com/timothybrooks/instruct-pix2pix/main/imgs/example.jpg"
+
+def download_image(url):
+    image = PIL.Image.open(requests.get(url, stream=True).raw)
+    image = PIL.ImageOps.exif_transpose(image)
+    image = image.convert("RGB")
+    return image
+
+image = download_image(URL)
```
Comment (abi) [inline README.md:76]: should be `url`

File: README.md [hunk 19]
```diff
@@ -36,6 +40,49 @@ python edit_app.py
 
 _(For advice on how to get the best results by tuning parameters, see the [Tips](https://github.com/timothybrooks/instruct-pix2pix#tips) section)._
 
+### 🧨 Diffusers
+
+1. Install diffusers and relevant dependencies:
+
+```bash
+pip install transformers accelerate torch
```
Comment (abi) [inline README.md:48]: the code snippet below allows need pillow and requests, should we include that here?

**Review by abi — COMMENTED** [review]
Thanks for adding this model to diffusers! It's much faster :)

**Comment by holynski** [comment]
Taking a look.

File: README.md (hunk 19 above)
Comment (patrickvonplaten) [reply README.md:48]: It's automatically installed when installing diffusers, so I think we're good :-) 

The other packages are soft dependencies in diffusers that help a lot with speed though and are necessary for this specific model

File: README.md (hunk 18 above)
Comment (patrickvonplaten) [reply README.md:76]: Thanks!

Comment (patrickvonplaten) [inline README.md:76]:
```suggestion
image = download_image(url)
```

**Comment by holynski** [comment]
Hey, thanks for adding this. 

We had a couple requests to add different alternative approaches, so we did a consolidation pass on this and #24  to add a section for other ways of using the model. Can you PTAL at HEAD and let me know if it looks okay?

**Comment by patrickvonplaten** [comment]
Sure, I adapted the PR :-) Happy to adapt some things if you like!

**Comment by holynski** [comment]
Hey, I think most of the content of your PR should now be covered in the updated Readme section. Can you take another look? 

If there's something in specific that seems to be missing or you'd like to modify, let me know, I'm happy to pull it in.

**Comment by patrickvonplaten** [comment]
Cool, yeah this section is nice: https://github.com/timothybrooks/instruct-pix2pix#instructpix2pix-in--diffusers
